### DISCUSSION TITLE
How to update values dynamically during simulation based on a condition

### DISCUSSION THREAD

**Discussion by TheASingh** [CURENT/andes#263] [Q&A]
Good morning Dr. Cui,

I'm working on **IEEE14 DGPRCT1** use case.
I have tested the Alter command, I think this is a manual change during simulation.

Now, my scenario is, I need to alter values dynamically based on a **certain condition**. I also went through the documentation and found **Alter model** under **TimedEvent group**.

If **Alter model** is used to update the values dynamically based on a **certain condition**, then it would be really helpful, If you can provide me any example to do that.

Regards,
A Singh

**Comment by cuihantao** [comment]
Alter is for predefined /static alteration. 

For user-defined conditions, you will need to write a script to implement that condition and then call the alter method on the parameter you want to change.

**Reply by TheASingh** [reply]
Dr. Cui,

Is there any way, with which I can check my condition during simulation?

As you wrote, I have to write a script. 
**For example**: In the script, I will be only able to check my condition after running the simulation for (let's say) 5 seconds or 10 seconds. 

I need to **check my condition and alter a parameter** while the ANDES is simulating. 
Your help will be appreciable.

Regards,
A Singh

**Reply by TheASingh** [reply]
Dr. Cui,

Let me clarify my use case in another way:

**1. Simulation should be running continuously.**
**Query 1**: Please guide me to achieve this.

**2. If I'm able to run the simulation continuously, then I want to apply a user-defined condition at any time. E.g. I want to alter the generation of a GENROU.**
**Query 2**: Is it possible? If yes, then please guide me to achieve this.

Your help will be really appreciable.

Regards,
A Singh

**Reply by cuihantao** [reply]
There is a so-called perturbation file that gets executed at each step. See https://github.com/cuihantao/andes/blob/7da7d4d1ac814a3f41266c85bea14fa526b38770/tests/test_paths.py#L38 and check the per file in `cases/ieee14/`.

You can implement your logic in the pert file with access to any elements in `System`.

**Reply by TheASingh** [reply]
Thanks Dr. Cui,

I was able to implement the logic in the pert file. 

Apart from that, **is there any way to update a value during simulation?**
Let's assume, simulation is already running for 30 mins, I want to update a value at anytime (e.g. at 15 min or 15.10 min) during simulation. In other words, user should be able to update a value at his desired time.

Regards,
A Singh
